mlkem/poly.c

```diff
@@ -338,6 +339,39 @@ void poly_getnoise_eta1(poly *r, const uint8_t seed[KYBER_SYMBYTES], uint8_t non
     poly_cbd_eta1(r, buf);
 }
 
+/*************************************************
+* Name:        poly_getnoise_eta1_4x
+*
+* Description: Batch sample four polynomials deterministically from a seed and nonces,
+*              with output polynomials close to centered binomial distribution
+*              with parameter KYBER_ETA1
+*
+* Arguments:   - poly *r{0,1,2,3}: pointer to output polynomial
+*              - const uint8_t *seed: pointer to input seed
+*                                     (of length KYBER_SYMBYTES bytes)
+*              - uint8_t nonce{0,1,2,3}: one-byte input nonce
+**************************************************/
+void poly_getnoise_eta1_4x(poly *r0,
+                           poly *r1,
+                           poly *r2,
+                           poly *r3,
+                           const uint8_t seed[KYBER_SYMBYTES],
+                           uint8_t nonce0,
+                           uint8_t nonce1,
+                           uint8_t nonce2,
+                           uint8_t nonce3)
+{
+    uint8_t buf[KECCAK_WAY][KYBER_ETA1 *KYBER_N / 4];
+    prf(buf[0], sizeof(buf[0]), seed, nonce0);
+    prf(buf[1], sizeof(buf[1]), seed, nonce1);
+    prf(buf[2], sizeof(buf[2]), seed, nonce2);
+    prf(buf[3], sizeof(buf[3]), seed, nonce3);
+    poly_cbd_eta1(r0, buf[0]);
+    poly_cbd_eta1(r1, buf[1]);
+    poly_cbd_eta1(r2, buf[2]);
+    poly_cbd_eta1(r3, buf[3]);
```

Comment:
These four calls to `prf` should be replaced with a single call to `shake256x4`?

Reply:
fixed in https://github.com/pq-code-package/mlkem-c-aarch64/pull/62/commits/6c716d18f1aada08abcd445c905fb77937dce1d5